janusgraph-hbase-parent/janusgraph-hbase-10/src/main/java/org/janusgraph/diskstorage/hbase/HBaseAdmin1_0.java

```diff
@@ -47,21 +53,29 @@ public void clearTable(String tableString, long timestamp) throws IOException
             return;
         }
 
-        if (!adm.isTableDisabled(tableName))
-            adm.disableTable(tableName);
-
-        if (!adm.isTableDisabled(tableName))
-            throw new RuntimeException("Unable to disable table " + tableName);
-
-        // This API call appears to both truncate and reenable the table.
-        log.info("Truncating table {}", tableName);
-        adm.truncateTable(tableName, true /* preserve splits */);
-
-        try {
-            adm.enableTable(tableName);
-        } catch (TableNotDisabledException e) {
-            // This triggers seemingly every time in testing with 1.0.2.
-            log.debug("Table automatically reenabled by truncation: {}", tableName, e);
+        // Unfortunately, linear scanning and deleting rows is faster in HBase when running integration tests than
+        // disabling and deleting/truncating tables.
+        final Scan scan = new Scan();
+        scan.setBatch(100);
+        scan.setCacheBlocks(false);
+        scan.setCaching(2000);
+        scan.setTimeRange(0, Long.MAX_VALUE);
+        scan.setMaxVersions(1);
+
+        final List<Delete> deleteList = new ArrayList<>();
+        final int batchSize = 100;
+        try (final Table table = adm.getConnection().getTable(tableName);
+             final ResultScanner scanner = table.getScanner(scan)) {
+            for (Result result = scanner.next(); result != null; ) {
+                final Delete delete = new Delete(result.getRow());
+                delete.setTimestamp(timestamp);
```

Comment:
I don't think it's necessary but it might be more correct. The timestamp is a method argument and it is set on the delete in the [HBase098 implementation](https://github.com/JanusGraph/janusgraph/blob/84ead7d907b62061e30c5d3210e5e8422fd4648c/janusgraph-hbase-parent/janusgraph-hbase-098/src/main/java/org/janusgraph/diskstorage/hbase/HBaseAdmin0_98.java#L87).